component/src/main/java/io/siddhi/extension/io/cdc/source/polling/strategies/WaitOnMissingRecordPollingStrategy.java

```diff
@@ -145,13 +151,26 @@ public void poll() {
                         handleEvent(detailsMap);
                     }
                 } catch (SQLException e) {
+                    if (metrics != null) {
+                        isError = true;
```

Comment:
set the isError variable outside the metrics null check

Reply:
Same as the above reason